### PR TITLE
Allow modification of namelist_args for all kinds of nested dictionaries.

### PR DESCRIPTION
**Pull request by ilopezgp**
This PR should allow changing any nested dictionary through namelist_args. I added a few options to the tests to make sure of it.

### CODE REVIEW

File: src/helper_funcs.jl [hunk 1]
```diff
@@ -342,3 +342,29 @@ function get_entry(dict, key, default)
...
+        else
+            return dict
+        end
+    end
+    # Unpack
+    value = keys_and_value[end]
+    keys = keys_and_value[1:(end - 1)]
+    # Modify entry
+    last_dict = get_last_nested_dict(dict, keys)
+    last_dict[keys[end]] = value
```
Comment (haakon-e) [inline src/helper_funcs.jl:369]:
```suggestion
    keys = keys_and_value[1:end-1]
    value = keys_and_value[end]
    subdict = dict
    for k in keys[1:end-1]
        subdict = subdict[k]
    end
    subdict[keys[end]] = value
```
alternative approach, see #168.

**Review by haakon-e — APPROVED** [review]
LGTM

File: src/helper_funcs.jl (hunk 1 above)
Comment (ilopezgp) [reply src/helper_funcs.jl:369]: Recursion vs iteration, it should be the same haha. I will keep it as is just for convenience.

**Pull request merged**